### PR TITLE
Bug 1836380  – adding strings to be used by new search settings screen in advance

### PR DESCRIPTION
**Pull request by mavduevskiy**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1836380

### CODE REVIEW

**Review by HarrisonOg — APPROVED** [review]

File: fenix/app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -346,6 +348,14 @@
...
     <!-- Preference category for all links about Fenix -->
     <string name="preferences_category_about">About</string>
+    <!-- Preference category for settings related to changing the default search engine -->
+    <string name="preferences_category_select_default_search_engine" tools:ignore="UnusedResources">Select one</string>
+    <!-- Preference for settings related to managing search shortcuts for the quick search menu -->
+    <string name="preferences_manage_search_shortcuts" tools:ignore="UnusedResources">Manage search shortcuts</string>
+    <!-- Preference category for settings related to managing search shortcuts for the quick search menu -->
+    <string name="preferences_category_engines_in_search_menu" tools:ignore="UnusedResources">Engines visible on the search menu</string>
+    <!-- Summary for preference for settings related to managing search shortcuts for the quick search menu -->
+    <string name="summary_preferences_manage_search_shortcuts" tools:ignore="UnusedResources">Edit engines visible in the search menu</string>
```
Comment (gabrielluong) [inline fenix/app/src/main/res/values/strings.xml:358]:
```suggestion
    <string name="preferences_manage_search_shortcuts_summary" tools:ignore="UnusedResources">Edit engines visible in the search menu</string>
```

Comment (gabrielluong) [reply fenix/app/src/main/res/values/strings.xml:358]: Maybe move this below `preferences_manage_search_shortcuts` as well

File: fenix/app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1810,14 +1822,24 @@
 
     <!-- Text for the button to create a custom search engine on the Add search engine screen -->
     <string name="search_add_custom_engine_label_other">Other</string>
+    <!-- Label for the TextField in which user enters custom search engine name -->
+    <string name="search_add_custom_engine_name_label" tools:ignore="UnusedResources">Name</string>
     <!-- Placeholder text shown in the Search Engine Name TextField before a user enters text -->
     <string name="search_add_custom_engine_name_hint">Name</string>
+    <!-- Placeholder text shown in the Search Engine Name TextField before a user enters text -->
```
Comment (gabrielluong) [inline fenix/app/src/main/res/values/strings.xml:1829]:
```suggestion
    <!-- Placeholder text shown in the Search Engine Name text field before a user enters text -->
```
Let's move away with the technical Android term here even though it already exists in other places. I am sure our translators are just reading it as "text field" anyways.

**Comment by gabrielluong** [comment]
Left some comments that might be worth considering, but no re-review needed from me.

**Pull request merged**